src/convertToObject.js

```diff
@@ -3,16 +3,42 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ *  (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const objectWithCssParams = {};
+
+  sourceString
+    .split(/\n/)
+    .filter((param) => {
+      const trimmedParam = param.trim();
+
+      return !(trimmedParam.length <= 1);
+    })
+    .map((param) => {
+      return param.split(':').map((value) => {
+        const fixedValue = value.includes(';')
+          ? value.slice(0, value.indexOf(';'))
+          : value;
+
+        return fixedValue.trim();
+      });
+    })
+    .reduce((accumulator, param) => {
+      accumulator[param[0]] = param[1];
+
+      return accumulator;
+    }, objectWithCssParams);
```

Comment:
Good idea to use `reduce`, but you could process every item inside this method and get rid of other methods like

> return sourceString.split(";").reduce((result, style) => {
>     there you get array from the whole string style.split(":");
>     there you trim every pair key = yourPair[0]?.trim();
>     there you trim every pair value = yourPair[1]?.trim();
>     if (key && value) {
>       result[key] = value;
>     }
>     return result;
>   }, {});

Reply:
ESLint forbids "?" before trim :(